NPE thrown when computing Code Actions on invalid code

Given :
```java
public class App {
  public static void main(String[] args) {
    App.test(); // compilation error, test() doesn't exist
  }
}
```
placing the cursor anywhere on the `App` word causes  textDocument/codeAction to fail

```
[Error - 4:25:17 PM] Request textDocument/codeAction failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.IMethod.getCompilationUnit()" because "iMethod" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.IMethod.getCompilationUnit()" because "iMethod" is null
	at org.eclipse.jdt.internal.corext.refactoring.code.makestatic.InitialConditionsChecker.checkSourceAvailable(InitialConditionsChecker.java:181)
	at org.eclipse.jdt.internal.corext.refactoring.code.MakeStaticRefactoring.checkInitialConditions(MakeStaticRefactoring.java:127)
	at org.eclipse.jdt.ls.core.internal.corrections.RefactorProcessor.getMakeMethodStaticRefactoringProposal(RefactorProcessor.java:1189)
	at org.eclipse.jdt.ls.core.internal.corrections.RefactorProcessor.getProposals(RefactorProcessor.java:228)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler.getCodeActionCommands(CodeActionHandler.java:221)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$15(JDTLanguageServer.java:779)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	... 6 more
```

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/RefactorProcessor.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/RefactorProcessor.java
@@ -79,6 +79,7 @@ import org.eclipse.jdt.internal.corext.fix.ConvertLoopFixCore;
 import org.eclipse.jdt.internal.corext.fix.IProposableFix;
 import org.eclipse.jdt.internal.corext.fix.LambdaExpressionsFixCore;
 import org.eclipse.jdt.internal.corext.fix.LinkedProposalModelCore;
+import org.eclipse.jdt.internal.corext.refactoring.Checks;
 import org.eclipse.jdt.internal.corext.refactoring.RefactoringAvailabilityTesterCore;
 import org.eclipse.jdt.internal.corext.refactoring.RefactoringCoreMessages;
 import org.eclipse.jdt.internal.corext.refactoring.code.ConvertAnonymousToNestedRefactoring;
@@ -103,6 +104,7 @@ import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
 import org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler;
 import org.eclipse.jdt.ls.core.internal.preferences.PreferenceManager;
 import org.eclipse.jdt.ls.core.internal.text.correction.ActionMessages;
+import org.eclipse.jdt.ls.core.internal.text.correction.CodeActionUtility;
 import org.eclipse.jdt.ls.core.internal.text.correction.RefactorProposalUtility;
 import org.eclipse.jdt.ls.core.internal.text.correction.RefactoringCorrectionCommandProposal;
 import org.eclipse.jdt.ui.cleanup.CleanUpContext;
@@ -1184,9 +1186,30 @@ public class RefactorProcessor {
 	}
 
 	private boolean getMakeMethodStaticRefactoringProposal(CodeActionParams params, IInvocationContext context, ASTNode coveringNode, ArrayList<ProposalKindWrapper> proposals, IProgressMonitor monitor) {
-		MakeStaticRefactoring refactoring = new MakeStaticRefactoring(context.getCompilationUnit(), context.getSelectionOffset(), context.getSelectionLength());
+		ASTNode methodNode = CodeActionUtility.inferASTNode(coveringNode, MethodDeclaration.class);
+		if (!(methodNode instanceof MethodDeclaration methodDeclaration)) {
+			return false;
+		}
+		IMethodBinding methodBinding = methodDeclaration.resolveBinding();
+		if (methodBinding == null) {
+			return false;
+		}
+		IJavaElement element = methodBinding.getJavaElement();
+
 		try {
-			if (refactoring != null && refactoring.checkInitialConditions(new NullProgressMonitor()).isOK()) {
+			if (!Checks.isAvailable(element)) {
+				return false;
+			}
+		} catch (JavaModelException e) {
+			return false;
+		}
+		if (!(element instanceof IMethod method)){
+			return false;
+		}
+
+		MakeStaticRefactoring refactoring = new MakeStaticRefactoring(method);
+		try {
+			if (refactoring.checkInitialConditions(monitor).isOK()) {
 				String label = RefactoringCoreMessages.MakeStaticRefactoring_name;
 				int relevance = IProposalRelevance.CHANGE_METHOD_SIGNATURE;
 				ChangeCorrectionProposalCore proposal = new ChangeCorrectionProposalCore(label, null /*create.getChange()*/, relevance) {
@@ -1210,5 +1233,4 @@ public class RefactorProcessor {
 		}
 		return true;
 	}
-
 }

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerTest.java
@@ -387,6 +387,30 @@ public class CodeActionHandlerTest extends AbstractCompilationUnitBasedTest {
 	}
 
 	@Test
+	public void testCodeAction_NPEOnInvalidMethodCall() throws Exception {
+		ICompilationUnit unit = getWorkingCopy(
+				"src/App.java",
+				"""
+				public class App {
+					String foo = App.test(); // compilation error, test() doesn't exist
+				}
+				"""
+				);
+		CodeActionParams params = new CodeActionParams();
+		params.setTextDocument(new TextDocumentIdentifier(JDTUtils.toURI(unit)));
+		String source = unit.getSource();
+		int appIndex = source.indexOf("App.test()");
+		final Range range = JDTUtils.toRange(unit, appIndex, "App".length());
+		params.setRange(range);
+		CodeActionContext context = new CodeActionContext(Collections.emptyList());
+		params.setContext(context);
+		// This should not throw NPE
+		List<Either<Command, CodeAction>> codeActions = getCodeActions(params);
+		Assert.assertNotNull("Code actions should not be null", codeActions);
+		// Code actions may be empty, but should not throw an exception
+	}
+
+	@Test
 	public void testCodeAction_refreshDiagnosticsCommandInNewCUProposal() throws Exception {
 		when(clientPreferences.isResolveCodeActionSupported()).thenReturn(true);
 		preferences.setValidateAllOpenBuffersOnChanges(false);

